cEP-0024.md

```diff
@@ -0,0 +1,85 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the details of the process of creating new automations and
+plugins in GitMate which helps coala development workflow, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there area number of plugins which saves a lot of time of our
+community(for example automatically close PRs from master). Many of them are
+general plugins, which are required by every org, but we need more plugins
+which support coala development workflow.
+
+## New Features
+
+### Auto-assign issue based on commit mention
+
+When a commit mentions(fixes, closes) a unassigned issues, auto assign
+commit author to the mentioned unassigned issues.
+
+### Auto-reject PR opened on master of forks
+
+Normally, we should make changes in different branch from master of forks, and
+submit a PR. But, sometimes we receive PRs from master of forks. To solve this
+problem, we need a mechanism by which gitmate automatically rejects those PR’s,
+and inform the PR author to work on a different branch.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he either approves changes or request changes or
+comments something and leaves it there, but PR would be still in the review
+queue, so there should be an automation to mark approve label if reviewer
+approves changes and mark WIP if reviewer requests changes or comments on the
+source code when there is CI failure.
+
+### Fixes vs Closes automated checking
+
+Our members often confuse when we should use `Fixes` or `Closes` on a commit.
+So, GitMate should automatically check every commit for issues which is being
+Fixed or Closed, if those issues have bug label(label will be configurable),
+`Fixes` should be used, and vice versa. If there is an error, gitmate commit
+review should fail and inform the PR author how to fix it.
```

Comment:
how should they fix it?
amend the commit message?

we need to have a custom message to link them to coala documentation for git commit message guidelines and git help.

How will gitmate parse the commit message to determine fixed/closed?  Or does GitHub provide this as metadata?
If a parser is needed, is it only going to support coala's commit message layout?  Will the commit parser be part of IGitt or gitmate?

Note that some of this *may* also be part of the commit project : https://github.com/coala/cEPs/pull/114

And there is a decent prototype at https://github.com/coala/coala-bears/pull/2416

Also ... fyi ... we may get better labelling happening if we have different templates for fixes vs features.  c.f. https://github.com/coala/coala/pull/5468

Reply:
We already have a IGitt parser to find fixes and closes issues, right now it checks only in a PR, not for a single commit, I am working on that.